### ISSUE TITLE
Error with geo_shape relation CONTAINS

### ISSUE THREAD

**Issue by glascaleia** [elastic/elasticsearch#17866]
Hi all.
I have the following error trying to execute a spatial query with a ShapeRelation Contain.
aused by: java.lang.IllegalArgumentException: 
	at org.elasticsearch.index.query.GeoShapeQueryParser.getArgs(GeoShapeQueryParser.java:192)
	at org.elasticsearch.index.query.GeoShapeQueryParser.parse(GeoShapeQueryParser.java:169)
	at org.elasticsearch.index.query.QueryParseContext.parseInnerQuery(QueryParseContext.java:250)
	at org.elasticsearch.index.query.QueryParseContext.parseInnerFilter(QueryParseContext.java:263)
	at org.elasticsearch.index.query.IndexQueryParserService.parseInnerFilter(IndexQueryParserService.java:220)
	at org.elasticsearch.search.query.PostFilterParseElement.parse(PostFilterParseElement.java:33)
	at org.elasticsearch.search.SearchService.parseSource(SearchService.java:838)
	at org.elasticsearch.search.SearchService.createContext(SearchService.java:654)
	at org.elasticsearch.search.SearchService.createAndPutContext(SearchService.java:620)
	at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:371)

Regards

**Comment by eskibars** [comment]
@glascaleia can you please post the query you're running that's giving you this error?

**Comment by glascaleia** [comment]
{
  "query" : {
    "bool" : {
      "must" : {
        "geo_shape" : {
          "mountainNowArea.location" : {
            "shape" : {
              "type" : "polygon",
              "coordinates" : [ [ [ -4.21875, 25.482951175 ], [ 15.1171875, 27.994401411 ], [ 24.609375, 18.312810846 ], [ 23.5546875, 7.710991655 ], [ -7.03125, 12.897489184 ], [ -4.21875, 25.482951175 ] ] ]
            },
            "relation" : "contains"
          },
          "_name" : null
        }
      }
    }
  },
  "sort" : [ {
    "mountainNowArea.creationDate" : {
      "order" : "desc"
    }
  } ]
}

**Comment by clintongormley** [comment]
This fails for me on 2.3.0, but works on master.  Full recreation:

    PUT t 
    {
      "mappings": {
        "t": {
          "properties": {
            "mountainNowArea": {
              "type": "object",
              "properties": {
                "location": {
                  "type": "geo_shape"
                },
                "creationDate": {
                  "type": "date"
                }
              }
            }
          }
        }
      }
    }
    
    PUT t/t/1
    {
      "mountainNowArea": {
        "location": {
          "type": "polygon",
          "coordinates": [
            [
              [
                100,
                0
              ],
              [
                101,
                0
              ],
              [
                101,
                1
              ],
              [
                100,
                1
              ],
              [
                100,
                0
              ]
            ]
          ]
        },
        "creationDate": "2015-01-01"
      }
    }
    
    GET _search
    {
      "query": {
        "bool": {
          "must": {
            "geo_shape": {
              "mountainNowArea.location": {
                "shape": {
                  "type": "polygon",
                  "coordinates": [
                    [
                      [
                        -4.21875,
                        25.482951175
                      ],
                      [
                        15.1171875,
                        27.994401411
                      ],
                      [
                        24.609375,
                        18.312810846
                      ],
                      [
                        23.5546875,
                        7.710991655
                      ],
                      [
                        -7.03125,
                        12.897489184
                      ],
                      [
                        -4.21875,
                        25.482951175
                      ]
                    ]
                  ]
                },
                "relation": "contains"
              },
              "_name": null
            }
          }
        }
      },
      "sort": [
        {
          "mountainNowArea.creationDate": {
            "order": "desc"
          }
        }
      ]
    }

**Comment by nknize** [comment]
@glascaleia what version of ES?

@clintongormley can you confirm the following exception?

```javascript 
"reason": "Failed to find geo_shape field [mountainNowArea.location]",
```
Can you confirm success when explicitly searching the `t` index? e.g.:

```javascript
GET t/_search
```

This failure is unrelated to the original IAE issue. So it sounds to me like there are 2 separate issues.

**Comment by clintongormley** [comment]
Fixed by #17866
